### PR TITLE
port archive-node build and archive-node test

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
This PR ports the archive node test to buildkite. Currently the test is failing on buildkite because node are not able to communicate with each other. In CircleCI I saw some similar logs, but we didn't save the entire log so I am not sure whether the test is running successfully on CI or not.

Additionally the archive node test is an integration level test. Maybe we would rewrite it in the future in the new framework.

### CODE REVIEW

**Comment by psteckler** [comment]
I think the changes to the `ppx_optcomp` and `ppx_version` submodules are not intended.

File: buildkite/scripts/setup-database-for-archive-node.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+user=$1
```
Comment (bkase) [inline buildkite/scripts/setup-database-for-archive-node.sh:1]: can you add to the top?

```bash
#!/bin/bash

set -eou pipefail
```

File: buildkite/src/Jobs/ArchiveNode.dhall [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+let JobSpec = ../Pipeline/JobSpec.dhall
+let Command = ../Command/Base.dhall
+let OpamInit = ../Command/OpamInit.dhall
+let Docker = ../Command/Docker/Type.dhall
+let Size = ../Command/Size.dhall
+in
+
+let user = "admin"
+let password = "codarules"
+let db = "archiver"
```
Comment (bkase) [inline buildkite/src/Jobs/ArchiveNode.dhall:14]: these variables are dead code now, right? Can we remove them?

File: buildkite/scripts/setup-database-for-archive-node.sh (hunk 1 above)
Comment (ghost-not-in-the-shell) [reply buildkite/scripts/setup-database-for-archive-node.sh:1]: Sure.

File: buildkite/src/Jobs/ArchiveNode.dhall (hunk 2 above)
Comment (ghost-not-in-the-shell) [reply buildkite/src/Jobs/ArchiveNode.dhall:14]: I am using these variables when I set up the database. I am passing these to the database script.

File: buildkite/scripts/setup-database-for-archive-node.sh (hunk 1 above)
Comment (bkase) [reply buildkite/scripts/setup-database-for-archive-node.sh:3]: Also can you add the `#!/bin/bash` to the first line?

**Pull request merged**